onadata/apps/api/viewsets/user_profile_viewset.py

```diff
@@ -95,6 +98,46 @@ def set_is_email_verified(profile, is_email_verified):
     profile.save()
 
 
+def check_user_lockout(request):
+    username = request.user.username
+    lockout = cache.get('{}{}'.format(LOCKOUT_CHANGE_PASSWORD_USER, username))
+    if lockout:
+        time_locked_out = \
+            datetime.now() - datetime.strptime(lockout,
+                                               '%Y-%m-%dT%H:%M:%S')
+        remaining_time = round(
+            (getattr(settings, 'LOCKOUT_TIME', 1800) -
+             time_locked_out.seconds) / 60)
+        response = {'error' :
+            'Too many password reset attempts, Try again in {} minutes'.format(
+                remaining_time)}
+        return response
+
+
+def change_password_attempts(request):
+    """Track number of login attempts made by user within a specified amount
+     of time"""
+    username = request.user.username
+    attempts = cache.get('{}{}'.format(CHANGE_PASSWORD_ATTEMPTS, username))
+
+    if attempts:
+        cache.incr('{}{}'.format(CHANGE_PASSWORD_ATTEMPTS, username))
+        attempts = cache.get('{}{}'.format(CHANGE_PASSWORD_ATTEMPTS, username))
+        if attempts >= getattr(settings, 'MAX_CHANGE_PASSWORD_ATTEMPTS', 10):
+            cache.set(
+                '{}{}'.format(LOCKOUT_CHANGE_PASSWORD_USER, username),
+                datetime.now().strftime('%Y-%m-%dT%H:%M:%S'),
+                getattr(settings, 'LOCKOUT_TIME', 1800))
```

Comment:
Considering `getattr(settings, 'LOCKOUT_TIME', 1800)` does not rely on any variable in the function, could we have it as a static variable on top of the page? We could use the same variable in all places this appears on this file, i.e.
```python
LOCKOUT_TIME = getattr(settings, 'LOCKOUT_TIME', 1800)
..
```
The same applies to `getattr(settings, 'MAX_CHANGE_PASSWORD_ATTEMPTS', 10)`.